calculator.html

```diff
@@ -0,0 +1,2178 @@
+          }
+          /* Generate Search String links for each IV set, formatted like: https://pvpivs.com/searchStr.html?mon=Togedemaru&cp=1500&r=15_f_f&set=10100*/
+          if (!filterMod) {
+            for (i = 0; i < rIV.length; i++) {
+              searchLink += "<p><a href='https://pvpivs.com/searchStr.html?mon=" + mon + "&cp=" + league + "&r=" + rIV[i] + "&set=10100'>Generate " + mon + " Search String to Trash below Rank " + rIV[i] + "</a></p>";
+            }
+          }
+        } else {
+          if (document.getElementById("dti").checked) {
+            trades = tradePcnts(ranks, keys, statProd, floor, false, false, false, false, league);
+          }
+        }
+
+        /* Print out link to custom IV Search String if any ranks were filtered out*/
+        if (filterMod) {
+          var searchMon = mon;
+          var value = pokeListObj[mon].split(',');
+          /* Trim trailing comma off string */
+          outIVs = outIVs.replace(/,\s*$/, "");
+          /* Ensure that mon is a base form, if not find the base form! */
+          if (!(value[0].includes("b"))) {
+            /* Not a base form, check evolutions */
+            var monFamilyNames = value.slice(4, value.length);
+            for (i = 0; i < monFamilyNames.length; i++) {
+              if ((pokeListObj[monFamilyNames[i]].split(",")[0]).includes("b")) {
+                /*console.log("Found the base mon! "+pokeListObj[monFamilyNames[i]].split(",")[0]+", setting searchMon="+monFamilyNames[i]);*/
+                searchMon = monFamilyNames[i];
+              }
+            }
+          }
+          searchLink += "<p><a href='https://www.pvpivs.com/searchStr.html?mon=" + searchMon + "&cp=" + league + "&o=000011&IVentry=" + outIVs + "'>Generate " + searchMon + " Search String for only these filtered IVs</a></p>";
+        }
+
+        document.getElementById('outputBox').innerHTML = evoHtml + evoFeedback + html + ranges + searchLink + "<br>" + trades;
+
+        /* Append how many total ranks were generated at the end of "Top <Ranks> <League> <Mon>" top line on page" */
+        document.getElementById("instructionsBox").innerHTML += "<p id='totalRanks'> of " + ranks.numRanks + ":</p><br />";
+
+        if (perfTiming) {
+          T1 = performance.now();
+          console.log(mon + " outputUserRanks: " + (T1 - T0).toFixed(1) + "ms (CMP " + (Tc - T0).toFixed(1) + "ms +IVs " + (Ti - Tc).toFixed(1) + "ms + Rows " + (Tr - Ti).toFixed(1) + "ms) for " + ranks.numRanks + " ranks.");
+          document.getElementById('timing_outputs').innerHTML += "outputUserRanks: " + (T1 - T0).toFixed(1) + "ms (CMP " + (Tc - T0).toFixed(1) + "ms +IVs " + (Ti - Tc).toFixed(1) + "ms + Rows " + (Tr - Ti).toFixed(1) + "ms) for " + ranks.numRanks + " ranks.<br>";
+        }
+        return [printLimit];
+      }
+
+      function filterAdjusted() {
+        filterMod = true;
+        main();
+      }
+
+      function main(mon) {
+
+        return;
```

Comment:
main() and all calls to main() should probably be taken out. I didn't go through all the code but as far as I could tell, main() isn't needed.